packages/headless/src/api/search/search/raw.ts

```diff
@@ -1,27 +1,21 @@
-type BaseRaw = {
+export interface Raw {
+  /**
+   * Custom keys that depend on the documents in the index.
+   */
+  [key: string]: unknown;
+
+  /**
+   * @internal @preapproved
+   */
   urihash: string;
-  parents: string;
-  sfid: string;
-  sfparentid: string;
-  sfinsertedbyid: string;
-  documenttype: string;
-  sfcreatedbyid: string;
+
+  /**
+   * @internal @preapproved
+   */
   permanentid: string;
-  date: number;
-  objecttype: string;
-  sourcetype: string;
-  sftitle: string;
-  size: number;
-  sffeeditemid: string;
-  clickableuri: string;
-  sfcreatedby: string;
-  source: string;
-  collection: string;
-  connectortype: string;
-  filetype: string;
-  sfcreatedbyname: string;
-  sflikecount: number;
-  language: string[];
-};
 
-export type Raw = BaseRaw & Record<string, unknown>;
+  /**
+   * @internal @preapproved
+   */
+  sourcetype: string;
+}
```

Comment:
**Breaking Change Context**

There are now only three properties: `urihash`, `permanentid` and `sourcetype`.

The interface also has an indexer: `[key: string]: unknown` to allow people to access keys without error. For example, the following will continue to work without any typescript error.

```
const template = `${result.raw.source}`
```

However, if the code was assuming a string, then there will be an error.

```
const source = result.raw.source.trim() //ts-error
``` 

**Why remove keys?**

The removed keys are not guaranteed to be returned. They depend on what is indexed. The old interface was not accurately reflecting the response object's shape.

**Less severe alternatives**

1. We could do nothing, and let people solve the error at runtime.
2. We could mark the fields as optional as suggested by Alexis. This provides autocomplete, while still allowing typescript to warn that it may be `undefined`. The downside is if we type `source` as a `string`, but it is a `number`, the interface is still wrong.

Reply:
Personally, in a perfect world where it doesn't cause issues with documentation or can easily be fixed, I would consider making `Result` generic. For instance:
```ts
export interface BaseFields {
  // urihash and stuff
}

export interface CommonFields extends BaseFields {
  [key: string]: unknown;
  ...
  // objecttype and stuff
}

export interface Result<RawFields extends BaseFields = CommonFields> {
  ...
  raw: RawFields;
}
```